threeML/utils/spectrum/spectrum_likelihood.py

```diff
@@ -0,0 +1,315 @@
+from threeML.utils.statistics.likelihood_functions import half_chi2
+from threeML.utils.statistics.likelihood_functions import poisson_log_likelihood_ideal_bkg
+from threeML.utils.statistics.likelihood_functions import poisson_observed_gaussian_background
+from threeML.utils.statistics.likelihood_functions import poisson_observed_poisson_background
+from threeML.exceptions.custom_exceptions import custom_warnings, NegativeBackground
+
+import copy
+import numpy as np
+
+
+
+
+# These classes provide likelihood evaluation to SpectrumLike and children
+
+
+
+class SpectrumLikelihood(object):
```

Comment:
very nice!

Reply:
although, maybe `BinnedStatistic` is a more appropriate name? `Likelihood` could be confused with `Like` that we use for plugins.

Reply:
yeah, I thought about that one. Will update.